lib/src/async_message_grouper.dart

```diff
@@ -3,50 +3,111 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'dart:async';
-import 'dart:collection';
+import 'dart:math';
+import 'dart:typed_data';
 
 import 'package:async/async.dart';
 import 'package:pedantic/pedantic.dart';
+import 'package:protobuf/protobuf.dart';
 
 import 'message_grouper.dart';
-import 'message_grouper_state.dart';
 
 /// Collects stream data into messages by interpreting it as
 /// base-128 encoded lengths interleaved with raw data.
 class AsyncMessageGrouper implements MessageGrouper {
-  /// Current state for reading in messages;
-  final _state = MessageGrouperState();
-
   /// The input stream.
   final StreamQueue<List<int>> _inputQueue;
 
-  /// The current buffer.
-  final Queue<int> _buffer = Queue<int>();
+  /// The current input buffer.
+  List<int> _buffer = [];
```

Comment:
Lets rename this to `_inputBuffer` and `_bufferPos` to `_inputBufferPos` so its a bit more self documenting